assets/js/components/StreamArea.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useEffect, useState } from "react";
+import ModePanel from "./ModePanel";
+import PresenterStreams from "./PresenterStreams";
+import HlsPlayer from "./HlsPlayer";
+import "../../css/streamarea.css";
+
+export type Mode = "presenters" | "hls";
+
+type StreamAreaProps = {
+  clientName: string;
+  eventChannel: any;
+};
+
+const StreamArea = ({ clientName, eventChannel }: StreamAreaProps) => {
+  const [mode, setMode] = useState<Mode>("hls");
+  const [hlsUrl, setHlsUrl] = useState<string>("");
+  const [presenterName, setPresenterName] = useState<string>("");
+
+  const addHlsUrl = (message: { name: string; playlist_idl: string }): void => {
+    const link = window.location.href.split("event")[0] + "video/";
+    if (message.playlist_idl) {
+      setHlsUrl(link + message.playlist_idl + "/video.m3u8");
```

Comment:
Use template literals.

Reply:
Fixed (tho I'm not the one that wrote that :( ).